### PR TITLE
docs: article about integrating WatchOS app in NativeScript project

### PR DESCRIPTION
**Pull request by NickIliev**
Related to https://github.com/NativeScript/nativescript-cli/issues/4541

### CODE REVIEW

File: docs/tooling/ios-watch-apps.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: iOS WatchOS Applications
+description: Use NativeScript to build your WatchOS apps that will bring users timely information and provide them with a way to interact on the go.
+position: 90
+slug: ios-watch-apps
+---
+
+# iOS WatchOS Applications
+
+From version 5.4 of the NativeScript CLI, an integration of [WatchOS](https://developer.apple.com/watchos/) application supports for your iOS mobile app created with NativeScript.
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md]: With version 5.4 the NativeScript CLI introduces a Beta support for integrating a [WatchOS](https://developer.apple.com/watchos/) application in  your iOS mobile app created with NativeScript.

**Comment by KristianDD** [comment]
Maybe it is a good idea to add a beta Note somewhere in the article.

File: docs/tooling/ios-app-extensions.md [hunk 2]
```diff
@@ -30,32 +30,30 @@ tns create today-extension --tsc
...
+    ```JSON
+    {
+        "frameworks": ["NotificationCenter.framework"],
+        "assetcatalogCompilerAppiconName": "AppIconExtension"
+    }
+    ```
+    **_`frameworks`_** key - the value is an array of framework names.
 
- `frameworks` key - the value is an array of framework names.
+    **_`assetcatalogCompilerAppiconName`_** key - the value is the icon file name located in `.xcassets` catalog with `AppIconExtension.appiconset` inside the extension folder.
```
Comment (KristianDD) [inline docs/tooling/ios-app-extensions.md]: the value is the name of the `.appiconset` you want to use from the `.xcassets` catalog inside the extension folder -(AppIconExtension.appiconset in the example above).

File: docs/tooling/ios-watch-apps.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Prerequisites
+
+- **NativeScript CLI version 5.4** and above.
+- NativeScript project.
+- **Xcode 10** and above.
+- Paired devices or simulators (**iPhone** with **iWatch** with **WatchOS 4.x/5.x** and above).
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md]: WatchOS 4.x won't work with the default Watch App created with Xcode. The user will have to manually set WATCHOS_DEPLOYMENT_TARGET in the configuration `json`.

File: docs/tooling/ios-watch-apps.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+**3.** Add a name to your watch app, for example, **_MyFirstWatchApp_**.
+
+**4.** Copy the generated **_MyFirstWatchApp_** and **_MyFirstWatchAppExtension_** to **_.../apps/MyApp/app/App_Resources/iOS/watchapp/MyFirstWatchApp_** and **_.../apps/MyApp/app/App_Resources/iOS/watchextension/MyFirstWatchAppExtension_** respectively.
+
+**5.** Inside the **_Info.plist_** of the Watch App replace the value of **_`WKCompanionAppBundleIdentifier`_** with **_`$(WK_APP_BUNDLE_IDENTIFIER)`_**.
+
+**6.** Inside the **_Info.plist_** of the Watch Extension replace the value of **_`WKAppBundleIdentifier`_** with **_`$(WK_APP_BUNDLE_IDENTIFIER)`_**.
+
+**7.** You can populate the **_Assets.xcassets_** of the Watch App and the Watch Extension and add the name of the **_`appiconset`_** to the **_.../apps/MyApp/app/App_Resources/iOS/watchapp/MyFirstWatchApp/watchapp.json_** and **_.../apps/MyApp/app/App_Resources/iOS/watchextension/MyFirstWatchApp Extension/extension.json_** respectively:
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md]: assetcatalogCompilerAppiconName is valid only for Watch App but not the Watch Extension.

The extension should support the targetBuildConfigurationPropertiy ASSETCATALOG_COMPILER_COMPLICATION_NAME, but I haven't tested that scenario.
   ```JSON
    {
        "targetBuildConfigurationProperties": {
            "ASSETCATALOG_COMPILER_COMPLICATION_NAME": "Complication"
        }
    }

File: docs/tooling/ios-watch-apps.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## WatchOS application in NativeScript.
+
+To integrate your existing WatchOS application into your NativeScript project, execute the following steps:
+
+**1.** Create **_Single View App_** from Xcode.
+
+**2.** Add watch app target through **_File > New > Target > WatchKit App_**.
+
+**3.** Add a name to your watch app, for example, **_MyFirstWatchApp_**.
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md]: We could add a step to check that Objective C is selected.

File: docs/tooling/ios-watch-apps.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+3. Add a name to your watch app, for example, **_MyFirstWatchApp_**. In the same screen verify that **Objective-C** is selected as your language.
+
+4. Copy the generated **_MyFirstWatchApp_** and **_MyFirstWatchAppExtension_** to **_.../apps/MyApp/app/App_Resources/iOS/watchapp/MyFirstWatchApp_** and **_.../apps/MyApp/app/App_Resources/iOS/watchextension/MyFirstWatchAppExtension_** respectively.
+
+5. Inside the **_Info.plist_** of the Watch App replace the value of **_`WKCompanionAppBundleIdentifier`_** with **_`$(WK_APP_BUNDLE_IDENTIFIER)`_**.
+
+6. Inside the **_Info.plist_** of the Watch Extension replace the 
+ of **_`WKAppBundleIdentifier`_** with **_`$(WK_APP_BUNDLE_IDENTIFIER)`_**.
+
+7. You can populate the **_Assets.xcassets_** of the Watch App and the Watch Extension and add the name of the **_`appiconset`_** to the **_.../apps/MyApp/app/App_Resources/iOS/watchapp/MyFirstWatchApp/watchapp.json_**:
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md]: Remove the Watch Extension from this section. Its iconset should be populated with using targetBuildConfigurationProperties:
```JSON
    {
        "targetBuildConfigurationProperties": {
            "ASSETCATALOG_COMPILER_COMPLICATION_NAME": "Complication"
        }
    }

File: docs/tooling/ios-watch-apps.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+- **Xcode 10** and above.
+- Paired devices or simulators (**iPhone** and **iWatch** with **WatchOS 4.x/5.x** and above).
+> **Note:** WatchOS 4.x won't work with the default Watch App created with Xcode. The user will have to manually set **_`WATCHOS_DEPLOYMENT_TARGET`_** in the configuration json (see below for details).
+
+- WatchOS app created with **Objective-C** (Swift code [is **not** supported yet](https://github.com/NativeScript/nativescript-cli/issues/4541#issuecomment-491202270)).
+
+
+## WatchOS application in NativeScript.
+
+To integrate your existing WatchOS application into your NativeScript project, execute the following steps:
```
Comment (KristianDD) [inline docs/tooling/ios-watch-apps.md:25]: If it is an existing WatchOS, the user should skip steps 1-3.

**Review by KristianDD — APPROVED** [review]

**Review by tsonevn — APPROVED** [review]

**Pull request merged**